aten/src/ATen/function_wrapper.py

```diff
@@ -1299,11 +1299,11 @@ def emit_body(env, option):
                 body.append(CodeTemplate(return_tensor).substitute(
                     env, wrapped_tensor=wrapped_tensor, maybe_scalar=maybe_scalar))
             # return the same underlying Tensor type for both real and accreal; this ensures
-            # e.g. x.sum(0) and x.sum() return the same type.
+            # e.g. x.sum(0) and x.sum() return the same type. We explicitly cast to the
+            # ScalarType before constructing the scalarTensor to avoid overflow checking.
```

Comment:
Mostly because some tests blindly call `x.sum()` on ByteTensors and trigger the check. This isn't great, but it matches the existing behavior of `sum(dim=0)`. Longer term I'd like to:

1) accept an optional dtype when calling sum()
2) probably default to a sufficiently large type (e.g. int64_t for ByteTensor)
3) handle mixed precision operations, at least with scalars, so that things like x / x.sum() don't break